### PR TITLE
Set ContentType to null if the response is NotModified

### PR DESCRIPTION
**Pull request by ajinkya-nikam**
Set ContentType to null when we set StatusCode to 304 (not Modified)

Addresses #8230

### CODE REVIEW

File: src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs [hunk 1]
```diff
@@ -75,6 +75,7 @@ internal enum PreconditionState
             if (preconditionState == PreconditionState.NotModified)
             {
                 response.StatusCode = StatusCodes.Status304NotModified;
+                response.ContentType = null;
```
Comment (asn19595) [inline src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs]: Set ContentType to null if the response is NotModified.

File: src/Mvc/Mvc.Core/test/FileStreamResultTest.cs [hunk 2]
```diff
@@ -437,6 +437,7 @@ public async Task WriteFileAsync_NotModified_RangeRequestedIgnored()
             Assert.Null(httpResponse.ContentLength);
             Assert.Empty(httpResponse.Headers[HeaderNames.ContentRange]);
             Assert.NotEmpty(httpResponse.Headers[HeaderNames.LastModified]);
+            Assert.Empty(httpResponse.Headers[HeaderNames.ContentType]);
```
Comment (asn19595) [inline src/Mvc/Mvc.Core/test/FileStreamResultTest.cs]: Check for ContentType not being set.

**Comment by mkArtak** [comment]
Thanks for your effort, @asn19595.
We have a process in place for accepting community PRs, and it starts from filing an issue. We engage into discussion with community members on the issues first to determine what are the problems the person is facing and what are the options for addressing it.
After finding out that the issue is worth fixing, then we move forward with submitting a PR.

Given that you already have a PR, please file an issue with the details of the problem and link it from this PR.

**Comment by asn19595** [comment]
> Thanks for your effort, @asn19595.
> We have a process in place for accepting community PRs, and it starts from filing an issue. We engage into discussion with community members on the issues first to determine what are the problems the person is facing and what are the options for addressing it.
> After finding out that the issue is worth fixing, then we move forward with submitting a PR.
> 
> Given that you already have a PR, please file an issue with the details of the problem and link it from this PR.

Sorry was new to the process. Addressed the issue that is already been filed by me. Thanks.

**Comment by pranavkm** [comment]
There's an ongoing PR that addresses this in a better way: https://github.com/aspnet/AspNetCore/pull/8557. Closing this in favor of the other.

**Pull request closed without merging**